### PR TITLE
Fix persistence recovery

### PR DESCRIPTION
**Pull request by kaikawaliu**
## Description

- Some changes on test_node_simple.sh for fixing some problem shown in local test 
- Fixed the logic when storing DSIncompleted in vacuous epoch
- Solved the ds->m_consensusID not matching after recovery

### Implementation

- [X] **ready for review**

### Integration Test (Core Team)

- [X] local machine test

- [x] small-scale cloud test with recovery for commit: dddd35a (not showing up due to rebase)
- [x] small-scale cloud test without recovery for commit: dddd35a (not showing up due to rebase)
- [ ] medium-scale (100) cloud test with recovery for commit cb71cb0

### CODE REVIEW

**Comment by Gnnng** [comment]
I'm helping to run the cloud test for this.

**Comment by bb111189** [comment]
[Help needed] Will be good to understand why there is an additional need for the sleep for ds leader. Is there any recent code change that causes this issue? If so, will it impact other area of the codebase

**Comment by kaikawaliu** [comment]
This may be the issue
```
[INFO   ][TID  7656][03:20:04:623][Prepare                       ] BEGIN
[INFO   ][TID  7656][03:20:04:623][UpdateDSBlockRand             ] BEGIN
[INFO   ][TID  7656][03:20:04:623][UpdateDSBlockRand             ] Genesis DSBlockchain
[INFO   ][TID  7656][03:20:04:623][UpdateDSBlockRand             ] END
[INFO   ][TID  7656][03:20:04:623][UpdateTxBlockRand             ] BEGIN
[INFO   ][TID  7656][03:20:04:623][UpdateTxBlockRand             ] Genesis txBlockchain
[INFO   ][TID  7656][03:20:04:623][UpdateTxBlockRand             ] END
[INFO   ][TID  7656][03:20:04:623][SetState                      ][Epoch 1] Node State is now POW_SUBMISSION at epoch 1
[INFO   ][TID  7656][03:20:40:126][Prepare                       ] END
```
 This is how the Prepare function looks like:

```
void Node::Prepare(bool runInitializeGenesisBlocks)
{
    LOG_MARKER();
    m_mediator.m_currentEpochNum
        = m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum()
        + 1;
    m_mediator.UpdateDSBlockRand(runInitializeGenesisBlocks);
    m_mediator.UpdateTxBlockRand(runInitializeGenesisBlocks);
    SetState(POW_SUBMISSION);
    POW::GetInstance().EthashConfigureLightClient(
        m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() + 1);
}
```

I'm afraid that `POW::GetInstance().EthashConfigureLightClient` is taking too much time

**Comment by Gnnng** [comment]
We probably should divide the cloud test into two:

1. a testnet running without recovery to make sure the changes don't affect the normal code path
2. a testnet running with recovery to verify that it indeed fixes the recovery issue (I've done this, and it's passed)

**Comment by ShengguangXiao** [comment]
A single EthashConfigureLightClient on the cloud node will takes around 1.8 seconds, but in the local test, 20 nodes will run EthashConfigureLightClient together, the CPU needs more time to finish all the EthashConfigureLightClient. Following is the log from cloud node test.
[INFO   ][TID    33][13:33:22:108][UpdateDSBlockRand             ] END
[INFO   ][TID    33][13:33:22:108][UpdateTxBlockRand             ] BEGIN
[INFO   ][TID    33][13:33:22:108][UpdateTxBlockRand             ] Genesis txBlockchain
[INFO   ][TID    33][13:33:22:108][UpdateTxBlockRand             ] END
[INFO   ][TID    33][13:33:22:108][SetState                      ][Epoch 1] Node State is now POW_SUBMISSION at epoch 1
[INFO   ][TID    33][13:33:23:956][Prepare                       ] END

File: tests/Node/test_node_simple.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ python tests/Zilliqa/test_zilliqa_local.py stop
 python tests/Zilliqa/test_zilliqa_local.py setup 20
 python tests/Zilliqa/test_zilliqa_local.py start 10
 
-sleep 30
+sleep 40
```
Comment (KaustubhShamshery) [inline tests/Node/test_node_simple.sh]: Is this a necessary change or side effect ??

**Pull request merged**